### PR TITLE
 txscript: Remove OP_SMALLDATA

### PR DESCRIPTION
**Pull request by dnldd**
This PR contains the following upstream commits:

- [f6cd49a](https://github.com/btcsuite/btcd/commit/f6cd49ac51964c016f2181291a67c8253e0f6a33)
  - NOOP
- [4494f0f](https://github.com/btcsuite/btcd/commit/4494f0f85257c5d1863831345a503864e4ac0c76)
  - updated `TestOpcodeDisasm` `OP_UNKNOWN#` range checks.

### CODE REVIEW

File: txscript/opcode_test.go [hunk 1]
```diff
@@ -141,8 +140,8 @@ func TestOpcodeDisasm(t *testing.T) {
 				expectedStr = "OP_NOP" + strconv.Itoa(int(val))
 			}
 
-		// OP_UNKNOWN#.
-		case opcodeVal >= 0xc1 && opcodeVal <= 0xf8 || opcodeVal == 0xfc:
+			// OP_UNKNOWN#.
```
Comment (davecgh) [inline txscript/opcode_test.go]: Seem like an extra tab snuck in here.

File: txscript/opcode_test.go [hunk 2]
```diff
@@ -207,8 +206,8 @@ func TestOpcodeDisasm(t *testing.T) {
...
```
Comment (davecgh) [inline txscript/opcode_test.go]: And here.

File: txscript/opcode.go [hunk 3]
```diff
@@ -586,9 +586,9 @@ var opcodeArray = [256]opcode{
 	OP_UNKNOWN246: {OP_UNKNOWN246, "OP_UNKNOWN246", 1, opcodeNop},
 	OP_UNKNOWN247: {OP_UNKNOWN247, "OP_UNKNOWN247", 1, opcodeNop},
 	OP_UNKNOWN248: {OP_UNKNOWN248, "OP_UNKNOWN248", 1, opcodeNop},
+	OP_UNKNOWN249: {OP_UNKNOWN249, "OP_UNKNOWN249", 1, opcodeNop},
```
Comment (davecgh) [inline txscript/opcode.go]: This would introduce a hard fork.  It needs to remain `opcodeInvalid`.

Comment (dnldd) [reply txscript/opcode.go]: Will be sure to assert modifications to upstream commits before putting up a PR. Sorry about that.

File: txscript/opcode_test.go [hunk 4]
```diff
@@ -141,9 +140,13 @@ func TestOpcodeDisasm(t *testing.T) {
 				expectedStr = "OP_NOP" + strconv.Itoa(int(val))
 			}
 
-		// OP_UNKNOWN#.
+			// OP_UNKNOWN#.
```
Comment (davecgh) [inline txscript/opcode_test.go]: There appears to be an extra tab here.

File: txscript/opcode_test.go [hunk 5]
```diff
@@ -141,9 +140,13 @@ func TestOpcodeDisasm(t *testing.T) {
...
 		case opcodeVal >= 0xc1 && opcodeVal <= 0xf8 || opcodeVal == 0xfc:
 			expectedStr = "OP_UNKNOWN" + strconv.Itoa(int(opcodeVal))
+
+		// OP_INVALID.
+		case opcodeVal == 0xf9:
```
Comment (davecgh) [inline txscript/opcode_test.go]: There is no need to do this.  Instead, just replace the `0xfa` entry in the map with `OP_INVALID249`.

File: txscript/opcode_test.go [hunk 6]
```diff
@@ -210,6 +213,9 @@ func TestOpcodeDisasm(t *testing.T) {
 		// OP_UNKNOWN#.
 		case opcodeVal >= 0xc1 && opcodeVal <= 0xf8 || opcodeVal == 0xfc:
 			expectedStr = "OP_UNKNOWN" + strconv.Itoa(int(opcodeVal))
+		// OP_INVALID.
+		case opcodeVal == 0xf9:
```
Comment (davecgh) [inline txscript/opcode_test.go]: Same here.

**Pull request merged**